packages/sdk/src/utils.ts

```diff
@@ -0,0 +1,86 @@
+import assert from 'assert'
+import {
+  Provider,
+  TransactionReceipt,
+  TransactionResponse,
+} from '@ethersproject/abstract-provider'
+import { getContractInterface } from '@eth-optimism/contracts'
+import { ethers } from 'ethers'
+import {
+  ProviderLike,
+  TransactionLike,
+  DirectionlessCrossChainMessage,
+} from './interfaces'
+
+/**
+ * Returns the canonical encoding of a cross chain message. This encoding is used in various
+ * locations within the Optimistic Ethereum smart contracts.
+ *
+ * @param message Cross chain message to encode.
+ * @returns Canonical encoding of the message.
+ */
+export const encodeCrossChainMessage = (
+  message: DirectionlessCrossChainMessage
```

Comment:
Using the `Directionless` version here because encoding it the same in both directions.